components/logins/android/src/main/java/mozilla/appservices/logins/DatabaseLoginsStorage.kt

```diff
@@ -280,3 +354,66 @@ internal fun Pointer.getAndConsumeRustString(): String {
 internal fun Pointer.getRustString(): String {
     return this.getString(0, "utf8")
 }
+
+/**
+ * A helper extension method for conveniently measuring execution time of a closure.
+ *
+ * N.B. since we're measuring calls to Rust code here, the provided callback may be doing
+ * unsafe things. It's very imporant that we always call the function exactly once here
+ * and don't try to do anything tricky like stashing it for later or calling it multiple times.
+ */
+inline fun <U> TimingDistributionMetricType.measure(funcToMeasure: () -> U): U {
+    val timerId = this.start()
+    try {
+        return funcToMeasure()
+    } finally {
```

Comment:
Should we be catching errors here too? I think so, since the counters are only on a few methods.

Reply:
AFAICT every timer measurement is inside a corresponding counter measurement, so *think* we should be OK here.